core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -155,15 +149,49 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
-    if (!format.equalsIgnoreCase("text")) {
-      try {
-        new Schema.Parser().parse(inputSchema);
-      } catch (Exception e){
-        throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    if (format.equalsIgnoreCase("text") && schema == null) {
+      return;
+    }
+    Schema parsedSchema;
+    try {
+      parsedSchema = Schema.parseJson(schema);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    }
+
+    if (format.equalsIgnoreCase("text") && parsedSchema != null) {
+      Schema.Field offsetField = parsedSchema.getField("offset");
+      if (offsetField == null) {
+        throw new IllegalArgumentException("Schema for text format must have an offset field");
+      }
+      Schema offsetSchema = offsetField.getSchema();
+      Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
+        offsetSchema.getType();
+      if (offsetType != Schema.Type.LONG) {
+        throw new IllegalArgumentException("Value of offset field in schema must be of type Long");
+      }
+
+      Schema.Field bodyField = parsedSchema.getField("body");
+      if (bodyField == null) {
+        throw new IllegalArgumentException("Schema for text format must have a body field");
+      }
+      Schema bodySchema = bodyField.getSchema();
+      Schema.Type bodyType = bodySchema.isNullable() ? bodySchema.getNonNullable().getType() : bodySchema.getType();
+      if (bodyType != Schema.Type.STRING) {
+        throw new IllegalArgumentException("Value of body field in schema must be of type String");
+      }
+    }
+
+    if (pathField != null) {
+      Schema.Field schemaPathField = parsedSchema.getField(pathField);
+      if (schemaPathField == null) {
+        throw new IllegalArgumentException("No value specified for " + pathField);
       }
-    } else if (inputSchema != null && !inputSchema.isEmpty()) {
-      if (!(inputSchema.contains("offset") && inputSchema.contains("body"))) {
-        throw new IllegalArgumentException("Schema must contain fields for offset and body");
+      Schema pathFieldSchema = schemaPathField.getSchema();
+      Schema.Type pathFieldType = pathFieldSchema.isNullable() ? pathFieldSchema.getNonNullable().getType() :
+        pathFieldSchema.getType();
+      if (pathFieldType != Schema.Type.STRING) {
+        throw new IllegalArgumentException("value of " + pathField + " must be of type String");
```

Comment:
Path field [whatever pathField is] must be of type string, but found [whatever pathFieldType is]